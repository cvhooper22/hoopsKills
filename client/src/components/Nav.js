import React from 'react';
import { NavLink } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

function Nav () {
  const { t } = useTranslation();

  return (
    <div className="nav-container">
      <div className="nav-item"><NavLink exact to="/boilerplate/main">{ t('components.nav.mainview') }</NavLink></div>
      <div className="nav-item"><NavLink exact to="/boilerplate/secondary">{ t('components.nav.secondaryview') }</NavLink></div>
      <div className="nav-item"><NavLink exact to="/boilerplate/featureflags">{ t('components.nav.featureflagsview') }</NavLink></div>
    </div>
  );
}

export default Nav;
