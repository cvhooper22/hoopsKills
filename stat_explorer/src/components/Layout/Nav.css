nav.nav {
  height: 3rem;
  box-shadow: 0 0.125rem 0.5rem rgb(0 0 0 / 12%);
  border-bottom: 1px solid rgba(0,0,0,.12);
}

nav li {
  font-weight: bold;
  font-size: 1rem;
  cursor: pointer;
}

nav li > sub {
  font-size: 0.5rem;
  text-transform: none;
  padding-left: 0.25rem;
  filter: opacity(0.4);
}

nav li:first-child {
  font-weight: bold;
  font-size: 1rem;
  cursor: pointer;
  padding-left: 0;
}

nav ul {
  list-style: none;
  padding-left: 0;
  margin: 0;
}

nav li:hover {
  filter: opacity(0.66);
}

.nav__item--disabled {
  color: gray;
  pointer-events: none;
}

.nav__item--active {
  color: var(--royal-blue);
}