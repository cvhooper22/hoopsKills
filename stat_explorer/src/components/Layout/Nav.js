import React from 'react';
import { useLocation } from 'react-router-dom';
import "./Nav.css";

export default function Nav ({ options = [], onOptionClick}) {
  const location = useLocation();
  const pathname = location.pathname;
  function handleClick (option) {
    return () => {
      if (onOptionClick) {
        onOptionClick(option);
      }
    }
  }

  return (
    <nav className='nav flex'>
      <ul className='flex-aic'>
        {
          options.map((opt) => {
            let isCurrent = pathname.includes(opt.routeRoot ?? opt.route);
            if (opt.route === '/') {
              isCurrent = pathname === opt.route;
            }
            const classes = ['px-l'];
            if (opt.disabled) {
              classes.push('nav__item--disabled');
            }
            if (isCurrent) {
              classes.push('nav__item--active')
            }
            return (
              <li className={classes.join(' ')} role="button" onClick={handleClick(opt)} key={opt.title}>
                {opt.title}
                {opt.disabled && <sub>Coming Soon</sub>}
              </li>
            );
          })
        }
      </ul>
    </nav>
  )
};
