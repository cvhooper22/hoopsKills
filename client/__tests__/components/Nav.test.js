import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { I18nextProvider } from 'react-i18next';

import Nav from '../../src/components/Nav';
import i18n from '../i18n';

describe('Nav -', () => {
  it('Should render Nav - Main view active', () => {
    const component = renderer.create(
      <I18nextProvider i18n={ i18n }>
        <MemoryRouter>
          <Nav
          ></Nav>
        </MemoryRouter>
      </I18nextProvider>
    );

    expect(component).toMatchSnapshot();
  });

  it('Should render Nav - Secondary view active', () => {
    const component = renderer.create(
      <I18nextProvider i18n={ i18n }>
        <MemoryRouter initialEntries={[ '/secondary' ]}>
          <Nav
          ></Nav>
        </MemoryRouter>
      </I18nextProvider>
    );

    expect(component).toMatchSnapshot();
  });

});