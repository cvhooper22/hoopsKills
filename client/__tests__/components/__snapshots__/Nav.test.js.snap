// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav - Should render Nav - Main view active 1`] = `
<div
  className="nav-container"
>
  <div
    className="nav-item"
  >
    <a
      aria-current={null}
      href="/boilerplate/main"
      onClick={[Function]}
    >
      Test - Main view
    </a>
  </div>
  <div
    className="nav-item"
  >
    <a
      aria-current={null}
      href="/boilerplate/secondary"
      onClick={[Function]}
    >
      Test - Secondary view
    </a>
  </div>
  <div
    className="nav-item"
  >
    <a
      aria-current={null}
      href="/boilerplate/featureflags"
      onClick={[Function]}
    >
      components.nav.featureflagsview
    </a>
  </div>
</div>
`;

exports[`Nav - Should render Nav - Secondary view active 1`] = `
<div
  className="nav-container"
>
  <div
    className="nav-item"
  >
    <a
      aria-current={null}
      href="/boilerplate/main"
      onClick={[Function]}
    >
      Test - Main view
    </a>
  </div>
  <div
    className="nav-item"
  >
    <a
      aria-current={null}
      href="/boilerplate/secondary"
      onClick={[Function]}
    >
      Test - Secondary view
    </a>
  </div>
  <div
    className="nav-item"
  >
    <a
      aria-current={null}
      href="/boilerplate/featureflags"
      onClick={[Function]}
    >
      components.nav.featureflagsview
    </a>
  </div>
</div>
`;
